x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ExpiredTokenRemover.java

```diff
@@ -69,10 +78,23 @@ public void doRun() {
                 }, this::onFailure));
     }
 
-    void submit(ThreadPool threadPool) {
+    boolean submit(ThreadPool threadPool) {
+        final List<String> indicesWithTokens = new ArrayList<>();
+        if (securityTokensIndex.isAvailable()) {
+            indicesWithTokens.add(securityTokensIndex.aliasName());
+        }
+        if (securityMainIndex.isAvailable() && (false == securityTokensIndex.indexExists()
+                || Instant.now().minus(MAXIMUM_TOKEN_LIFETIME_HOURS, ChronoUnit.HOURS).isBefore(securityTokensIndex.getCreationTime()))) {
+            indicesWithTokens.add(securityMainIndex.aliasName());
+        }
+        if (indicesWithTokens.isEmpty()) {
+            return false;
+        }
         if (inProgress.compareAndSet(false, true)) {
-            threadPool.executor(Names.GENERIC).submit(this);
+            threadPool.executor(Names.GENERIC).submit(() -> doRun(indicesWithTokens.toArray(new String[0])));
+            return true;
         }
+        return false;
     }
```

Comment:
`ExpiredTokenRemover` searches **both** indices and returns **true** only if it has run on at least one.
However it will **not** search (seek and destroy) tokens in `.security-7` **if** the `.security-tokens-7` was created more than 24h ago. This is because `.security-tokens-7` is created when **all** cluster nodes are upgraded to the new version, and this is when **all** nodes start to create and refresh on the new index. As I write this I've spotted a problem, when main security index might not be available at the last run. I will iterate on this.